handlers/update_resource.go

```diff
@@ -66,18 +87,48 @@ func (d *updateResourceEntry) mergeJSON(maps ...*datautils.JSONObject) datautils
 				} else {
 					result[k] = v
 				}
-			case json.Number:
-				// Cast "version" to int, otherswise dynamodbattribute.MarshalMap will cast it to String
-				// See https://github.com/aws/aws-sdk-go-v2/issues/115
-				i64, err := v.(json.Number).Int64()
-				if err != nil {
-					panic(err)
-				}
-				result[k] = int(i64)
 			default:
 				result[k] = v
 			}
 		}
 	}
 	return result
 }
+
+func (d *updateResourceEntry) buildNewResourceVersion(newResource *datautils.Resource, existingResource *datautils.Resource) {
+	newUUID, err := identifier.NewUUIDService().Mint()
+	if err != nil {
+		panic(err)
+	}
+
+	v, _ := newResource.JSON["version"].(json.Number).Float64()
+	newResource = datautils.NewResource(d.mergeJSON(&existingResource.JSON, &newResource.JSON)).
+		WithID(newUUID).
+		WithExternalIdentifier(existingResource.ExternalIdentifier()).
+		WithPrecedingVersion(existingResource.ID()).
+		WithVersion(int(v))
+
+	err = d.persistResource(newResource)
+	if err != nil {
+		panic(err)
+	}
+
+	depricatedResource := datautils.NewResource(existingResource.JSON).
+		WithFollowingVersion(newUUID)
+	//			WithCurrentVersion(false).
+
+	err = d.persistResource(depricatedResource)
+	if err != nil {
+		panic(err)
+	}
+}
+
+func (d *updateResourceEntry) verifyPayload(newResource *datautils.Resource, existingResource *datautils.Resource) error {
+	if newResource.ExternalIdentifier() != existingResource.ExternalIdentifier() {
+		return errors.New("Invalid ExternalIdentifier in payload")
```

Comment:
Can you be more specific in this message? Why is it invalid? Also, please use `externalIdentifier` as that is the correct property in the data.